k9mail-library/src/main/java/com/fsck/k9/mail/autoconfiguration/DNSHelper.java

```diff
@@ -0,0 +1,22 @@
+package com.fsck.k9.mail.autoconfiguration;
+
+
+import java.net.UnknownHostException;
+
+import org.xbill.DNS.MXRecord;
+import org.xbill.DNS.TextParseException;
+
+
+public class DNSHelper {
+    public static String getMXDomain(String domain) throws TextParseException, UnknownHostException {
```

Comment:
Should be `getMxDomain`. Also, it would be nicer if we didn't expose the TextParseException from the library to the caller here, a broken record might as well return `null` here.